### PR TITLE
[envoy-gw] Intergrate jwt feature with gRPC server

### LINKED ISSUE #1412: [envoy-gw] Need to implement a gRPC server for the envoy external auth service from java language
### Describe your problem(s)

As $subject, we need to implement a gRPC server for the envoy external auth filter from java language to make a connection between the external java filter chain and envoy proxy.

### Describe your solution

Use  [external_auth.proto ](https://github.com/envoyproxy/envoy/blob/master/api/envoy/service/auth/v3/external_auth.proto )  file to implement the gRPC server.

### How will you implement it

You can generate the required java code using [external_auth.proto ](https://github.com/envoyproxy/envoy/blob/master/api/envoy/service/auth/v3/external_auth.proto )  file or you can directly use already compiled stub dependency (https://mvnrepository.com/artifact/io.envoyproxy.controlplane/api).   

---

### PR DESCRIPTION
**Pull request by chashikajw**
### Purpose

As $subject, this will integrate jwt + cache feature with java gRPC server

### Issues

Fixes #1412

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [ ] Assigned 'Type' label
- [ ] Assigned the project
- [ ] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

**Review by Rajith90 — APPROVED** [review]

File: java-filter-chain/src/main/java/org/wso2/mgw/filterchain/JWTValidator/JWTValidator.java [hunk 1]
```diff
@@ -71,22 +76,32 @@ public String load(String s) throws Exception {
...
-            String key = (String) mapElement.getKey();
-            if (key == JWTConstants.AUTHORIZATION) {
-                valid = HandleJWT(request);
-                break;
-            }
+        CheckResponse response;
+
+        String token = headers.get(JWTConstants.AUTHORIZATION);
+        //Map<String, String> request = new HashMap<String, String>();
+        //request.put(JWTConstants.AUTHORIZATION, JWTConstants.JWT_TOKEN);
```
Comment (npamudika) [inline java-filter-chain/src/main/java/org/wso2/mgw/filterchain/JWTValidator/JWTValidator.java:85]: Shall we remove the commented code lines?

File: java-filter-chain/src/main/java/org/wso2/mgw/filterchain/JWTValidator/JWTValidator.java [hunk 2]
```diff
@@ -71,22 +76,32 @@ public String load(String s) throws Exception {
...
         }
-        return valid;
+
+        return response;
     }
     //handle JWT token
-    public static boolean HandleJWT(HashMap<String, String> requestAttributes){
-        String accessToken = requestAttributes.get(JWTConstants.AUTHORIZATION);
+    public static boolean HandleJWT(String accessToken){
+        //String accessToken = requestAttributes.get(JWTConstants.AUTHORIZATION);
```
Comment (npamudika) [inline java-filter-chain/src/main/java/org/wso2/mgw/filterchain/JWTValidator/JWTValidator.java:104]: Please remove the commented code line

File: java-filter-chain/src/main/java/org/wso2/mgw/filterchain/gRPC/server/ExtAuthService.java [hunk 3]
```diff
@@ -18,32 +18,46 @@
...
 
 public class ExtAuthService extends AuthorizationGrpc.AuthorizationImplBase {
 
     @Override
     public void check (CheckRequest request, StreamObserver<CheckResponse> responseObserver) {
 
-        System.out.println("++++++++++hit+++++++++++++++");
+        //System.out.println("++++++++++hit+++++++++++++++");
+
+        //System.out.println(request.getAttributes().getRequest().getHttp());
```
Comment (npamudika) [inline java-filter-chain/src/main/java/org/wso2/mgw/filterchain/gRPC/server/ExtAuthService.java:38]: Please remove the commented code lines

File: java-filter-chain/src/main/java/org/wso2/mgw/filterchain/gRPC/server/ExtAuthService.java [hunk 4]
```diff
@@ -18,32 +18,46 @@
...
 
-        System.out.println(request);
 
         // use a builder to construct a new Protobuffer object
         // jwt authentication should happens here
         //Status status = Status.OK;
```
Comment (npamudika) [inline java-filter-chain/src/main/java/org/wso2/mgw/filterchain/gRPC/server/ExtAuthService.java:42]: Please remove the commented code line

File: java-filter-chain/src/main/java/org/wso2/mgw/filterchain/gRPC/server/ExtAuthService.java [hunk 5]
```diff
@@ -18,32 +18,46 @@
...
-                .setStatus(Status.newBuilder().build())
+                .setStatus(Status.newBuilder().setCode(Code.OK_VALUE).build())
                 .setOkResponse(OkHttpResponse.newBuilder().build())
                 .build();
 
+
+        /*CheckResponse response1 = CheckResponse.newBuilder()
+                .setStatus(Status.newBuilder().setCode(Code.UNAUTHENTICATED_VALUE).build())
+                .setDeniedResponse(DeniedHttpResponse.newBuilder().build())
+                .build(); */
```
Comment (npamudika) [inline java-filter-chain/src/main/java/org/wso2/mgw/filterchain/gRPC/server/ExtAuthService.java:52]: Please remove the commented code lines

File: java-filter-chain/src/main/java/org/wso2/mgw/filterchain/gRPC/server/ExtAuthService.java [hunk 6]
```diff
@@ -18,32 +18,46 @@
...
+
+        Map<String, String> headers = request.getAttributes().getRequest().getHttp().getHeadersMap();
+
+
         // Use responseObserver to send a single response back
-        responseObserver.onNext(response);
+        CheckResponse response1 = JWTValidator.validateToken(headers);
+        //System.out.println("this is response "+ response1);
```
Comment (npamudika) [inline java-filter-chain/src/main/java/org/wso2/mgw/filterchain/gRPC/server/ExtAuthService.java:59]: Please remove the commented code lines

File: java-filter-chain/src/main/java/org/wso2/mgw/filterchain/JWTValidator/JWTValidator.java (hunk 1 above)
Comment (chashikajw) [reply java-filter-chain/src/main/java/org/wso2/mgw/filterchain/JWTValidator/JWTValidator.java:85]: This was removed

File: java-filter-chain/src/main/java/org/wso2/mgw/filterchain/JWTValidator/JWTValidator.java (hunk 2 above)
Comment (chashikajw) [reply java-filter-chain/src/main/java/org/wso2/mgw/filterchain/JWTValidator/JWTValidator.java:104]: This was removed

File: java-filter-chain/src/main/java/org/wso2/mgw/filterchain/gRPC/server/ExtAuthService.java (hunk 3 above)
Comment (chashikajw) [reply java-filter-chain/src/main/java/org/wso2/mgw/filterchain/gRPC/server/ExtAuthService.java:38]: This was removed

File: java-filter-chain/src/main/java/org/wso2/mgw/filterchain/gRPC/server/ExtAuthService.java (hunk 4 above)
Comment (chashikajw) [reply java-filter-chain/src/main/java/org/wso2/mgw/filterchain/gRPC/server/ExtAuthService.java:42]: This was removed

File: java-filter-chain/src/main/java/org/wso2/mgw/filterchain/gRPC/server/ExtAuthService.java (hunk 6 above)
Comment (chashikajw) [reply java-filter-chain/src/main/java/org/wso2/mgw/filterchain/gRPC/server/ExtAuthService.java:59]: This was removed

**Pull request merged**